File: server/src/main/java/com/springboot/store/entity/Staff.java
package com.springboot.store.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder

@Entity
@Table(name = "staff")
public class Staff implements UserDetails {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "email", nullable = false, unique=true)
    private String email;

    @Column(name = "password", nullable = false)
    private String password;

    @Column(name = "cccd")
    private String cccd;

    @Column(name = "address")
    private String address;

    @Column(name = "phone_number")
    private String phoneNumber;

    @Column(name = "facebook")
    private String facebook;

    @Column(name = "note")
    private String note;

    @Column(name = "sex")
    private String sex;

    @Column(name = "position")
    private String position;

    @Column(name = "salary_debt")
    private int salaryDebt = 0;

    @Column(name = "birthday")
    private Date birthday;

    @Column(name = "created_at")
    private Date createdAt;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "staff_salary_id")
    private StaffSalary staffSalary;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "avatar_id")
    private Media avatar;

    @OneToMany(mappedBy = "staff", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Token> tokens;

    @ManyToOne()
    @JoinColumn(name = "creator_id")
    private Staff creator;

    @ManyToOne()
    @JoinColumn(name = "staff_role_id")
    private StaffRole staffRole;

    @ManyToOne()
    @JoinColumn(name = "store_id")
    private Store store;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return staffRole.getAuthorities();
    }

    @Override
    public String getUsername() {
        return email;
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: entities/Staff.ts
import { SalarySetting } from "./SalarySetting";

export type Staff = {
  avatar: string;
  id: number;
  name: string;
  phoneNumber: string;
  cccd: string;
  salaryDebt: number;
  note: string;
  birthday: Date;
  sex: Sex;
  email: string;
  address: string;
  position: string;
  createAt: Date;
  role: string;
  salarySetting: SalarySetting;
  password: string | null;
};

export enum Sex {
  MALE = "Male",
  FEMALE = "Female",
}

export type Paycheck = {
  workingPeriod: string;
  totalSalary: number;
  paid: number;
  needToPay: number;
};

export type SimpleTransaction = {
  id: any;
  time: Date;
  description: string;
  value: number;
  salaryDebt: number;
};

export type WorkSchedule = {
  date: Date;
  shiftName: string;
  startTime: Date;
  endTime: Date;
  note: string;
};

export type Position = {
  id: any;
  name: string;
};
